code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm

```diff
@@ -506,115 +519,403 @@
+	var/maxdose = 0 // Internal value
+	var/totaldose = 0 //Internal value
+	var/messagedelay = ANTIDEPRESSANT_MESSAGE_DELAY
+	conflicting_reagents = list(/datum/reagent/alcohol/ = 5)
+
+/datum/reagent/antidepressants/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+
+	maxdose = max(maxdose,volume)
+
+	if(!ishuman(M) || world.time < data)
+		return
+	var/hastrauma = 0 //whether or not the brain has trauma
+	var/mob/living/carbon/human/H = M
+	var/obj/item/organ/brain/B = H.internal_organs_by_name["brain"]
+
+	if(B) //You won't feel anything if you don't have a brain.
+		for(var/x in B.traumas)
+			var/datum/brain_trauma/BT = x
+			var/goalvolume = cure_effects[BT]
+			if (goalvolume <= volume) // If the dosage is greater than the goal, then suppress the trauma.
+				if(!BT.suppressed)
+					BT.on_lose()
+					BT.suppressed = 1
+			else if(goalvolume-1 > volume) // -1 So it doesn't spam back and forth constantly if reagents are being metabolized
+				if(BT.suppressed)
+					BT.on_gain()
+					BT.suppressed = 0
+					hastrauma = 1
+
+		for(var/k in dosage_effects)
+			var/datum/brain_trauma/BT = k
+			var/percentchance = max(0,dosage_effects[k] - dose*10) // If you've been taking this medication for a while then side effects are rarer.
+			if(!H.has_trauma_type(BT) && prob(percentchance))
+				B.gain_trauma(BT,FALSE)
+
+		if(volume < maxdose*0.25) //If you haven't been taking youre regular dose, then cause issues.
+			H << "<span class='danger'>[pick(worstmessage)]</span>"
+			for(var/k in withdraw_effects)
+				var/datum/brain_trauma/BT = k
+				var/percentchance = max(withdraw_effects[k] * (dose/20)) //The highest the dosage, the more likely it is do get withdrawl traumas.
+				if(!H.has_trauma_type(BT) && prob(percentchance))
+					B.gain_trauma(BT,FALSE)
+		else if(hastrauma || volume < maxdose*0.5) //If your current dose is not high enough, then alert the player.
+			H << "<span class='warning'>[pick(badmessage)]</span>"
+		else
+			H << "<span class='good'>[pick(goodmessage)]</span>"
+
+	data = world.time + messagedelay
+
+/datum/reagent/antidepressants/affect_conflicting(var/mob/living/carbon/M, var/alien, var/removed, var/datum/reagent/conflicting)
+	M.hallucination = max(M.hallucination, 10)
+
+/datum/reagent/antidepressants/nicotine
+	name = "Nicotine"
+	id = "nicotine"
+	description = "Nicotine is a stimulant and relaxant commonly found in tobacco products. It is very poisonous, unless at very low doses."
+	reagent_state = LIQUID
+	color = "#333333"
+	metabolism = 0.002
+	overdose = 3
+	data = 0
+	taste_description = "bitterness"
+	goodmessage = list("You feel good.","You feel relaxed.","You feel alert and focused.")
+	badmessage = list("You start to crave nicotine...")
+	worstmessage = list("You need your nicotine fix!")
+	cure_effects = list(
+		/datum/brain_trauma/mild/phobia = 0.1,
+		/datum/brain_trauma/mild/muscle_weakness/ = 0.1
+	)
+	withdraw_effects = list(
+		/datum/brain_trauma/mild/muscle_weakness/ = 100
```

Comment:
Seems quite severe -> Would imho be better to use temporary modifiers for withdrawal.
Or add a way to make brain traumas temporary (i.e. expire after a certain time)
An alternative would be to add a "jittery" brain trauma that fades after a certain time